### ISSUE TITLE
508-defect-4 Enhancement [SCREENREADER, COGNITION]: Consider consolidating the information for the debt letter download links

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#12192]
# [508-defect-4 Enhancement](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-4)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

In a future iteration, consider providing the list of debt letters as a single link with the information in it, as shown in the enclosed screenshot. This will be a streamlined experience for screen reader users, and may reduce complexity.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user I want to understand the content of a link download without repetition of data or extra noise.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: staging

### Current code (address typo)

```diff
<table>
   <thead>
      <tr>
         <th>Date</th>
         <th>Type</th>
         <th>Action</th>
      </tr>
   </thead>
   <tbody>
      <tr>
         <td>May 29, 2020</td>
         <td>Second Demand Letter</td>
         <td>
           <a 
                download="Second Demand Letter dated May 29, 2020" 
                href="https://staging-api.va.gov/v0/debt_letters/%7B64B0BDC4-D40C-4C54-86E0-104C987B8D8F%7D">
                  <i aria-hidden="true" role="img" class="fas fa-download vads-u-padding-right--1"></i>
                  Download
                  <span class="sr-only">` $Second Demand Letter `</span> 
                  letter 
                  <span class="sr-only">`dated $May 29, 2020`</span> 
                  <dfn><abbr title="Portable Document Format">(PDF)</abbr></dfn>
           </a>
        </td>
      </tr>
<!-- etcetera -->
</table>
```

### Proposed code

```diff
<ol role="list">
  <li role="listitem">
    <a 
        href="https://staging-api.va.gov/v0/debt_letters/%7B64B0BDC4-D40C-4C54-86E0-104C987B8D8F%7D" 
        download="Second Demand Letter dated May 29, 2020">
          May 29, 2020 - Download Second Demand Letter 
          <dfn><abbr title="Portable Document Format">(PDF)</abbr></dfn>
    </a>
  </li>
<!-- etcetera -->
</ol>
```

### Current

![Screen Shot 2020-08-06 at 5 38 26 PM](https://user-images.githubusercontent.com/57469/89585365-b6b86880-d80b-11ea-817b-1895186b2818.png)

### Suggested

![Screen Shot 2020-08-06 at 3 12 54 PM](https://user-images.githubusercontent.com/57469/89584056-66400b80-d809-11ea-860d-4ab08f4d6eab.png)

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

**Comment by bradl3yC** [comment]
@jenstrickland your proposal is a complete design change from a table to an ordered list - I do not think we will be making this big of a shift at this point.  @rileyorr @rebeccawalsh @leahkeeler

**Comment by jenstrickland** [comment]
@bradl3yC I recognize that, which is why I said "In a future iteration." There wasn't an opportunity to collaborate with the designers on this MVP, and I don't know of another opportunity to provide this information.

**Comment by bradl3yC** [comment]
@jenstrickland can we just add the data to the screen reader only portion of the markup as opposed to changing the design for this section?

**Comment by jenstrickland** [comment]
@bradl3yC I don't understand the question. This issue ticket isn't suggesting making a change before launch. Launch as it is, apart from the typo noted in the other ticket.

**Comment by bradl3yC** [comment]
@jenstrickland the question is directed at the markup thats in place.. modifying the `<span class="sr-only">Second Demand Letter</span>` to include the full language like <span class="sr-only">May 29, 2020 - Download Second Demand Letter </span>`

**Comment by jenstrickland** [comment]
What we currently have is fine for an MVP.

If we did what is suggested here:

> @jenstrickland the question is directed at the markup thats in place.. modifying the `<span class="sr-only">Second Demand Letter</span>` to include the full language like `<span class="sr-only">May 29, 2020 - Download Second Demand Letter </span>`

That would cause the screen reader users to hear it all again. We'd suppress what sighted users see completely, and present a second experience for screen reader users. For those low-vision users who can see somewhat but use a screen reader as a support device, this disconnect would be confusing.

**Comment by jenstrickland** [comment]
Please document decisions made, and have the VSA accessibility specialist close the ticket.

**Comment by bradl3yC** [comment]
@jenstrickland your code suggested converts our table design to an ordered list.  I have already documented this earlier when this was originally proposed.  This is also a defect 4 enhancement and as such is also optional.  At this time, we're not changing our designs to an ordered or unordered list.  To my knowledge, there is no documentation, SOP or directive otherwise which mandates optional tickets be closed by anyone in particular.

**Comment by jenstrickland** [comment]
@bradl3yC I'm sorry this persists. I've let PMs know this repeatedly, and we have it documented at the bottom of issue tickets. Some older tickets lack this at the bottom. This is the process that is communicated in onboarding, reiterated in VSA frontend dev meeting and the design and product meetings. 
 
> Definition of done
> • Review and acknowledge feedback.
> • Fix and/or document decisions made.
> • Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

It isn't necessary to implement the change as an `ol`, but it is recommended that the designers consider the change to presentation. This ticket should be reviewed by @rileyorr and @rebeccawalsh for consideration in future work. Since this work moved directly into development without a dev intent review, this was documented as a consideration for a future iteration and not documented as a dev issue.

**Comment by bradl3yC** [comment]
@jenstrickland I was only made aware of the need for dev intent meetings yesterday - also, is the dev intent meeting _required_ by the VSP collaboration cycle?

**Comment by jenstrickland** [comment]
@bradl3yC That's disappointing. It's information that is generally shared in accessibility onboarding, and is in the VSA accessibility documentation. 

The dev intent meeting is not part of the current VSP collaboration cycle, although I have heard they may include it. The previous PM planned to add it. It is part of the VSA accessibility process.

**Comment by bradl3yC** [comment]
@jenstrickland can this be closed now?

**Comment by jenstrickland** [comment]
I would prefer this ticket be reassigned to @rileyorr and @rebeccawalsh and remain open as the work evolves.

**Comment by rebeccawalsh** [comment]
@jenstrickland We've weighed this feedback and would prefer to keep this data in a table, where data like date and document type can be sorted (as they are now since the screenshots above were taken.) In interest of our future document-centric work on eFolders, we'd like to maintain sorting and filtering as default functionality for document download tables.
